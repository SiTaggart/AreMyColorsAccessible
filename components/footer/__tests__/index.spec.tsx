/* eslint-env jest */
/// <reference types="jest" />

import { render } from '@testing-library/react';
import React from 'react';
import { Footer } from '..';

describe('Footer', (): void => {
  const mockProps = {
    styles: {
      footerLinks: {
        color: '#fff',
      },
    },
  };

  it('renders without crashing', (): void => {
    const { asFragment } = render(<Footer />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders footerLink styles added', (): void => {
    const { asFragment } = render(<Footer {...mockProps} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
