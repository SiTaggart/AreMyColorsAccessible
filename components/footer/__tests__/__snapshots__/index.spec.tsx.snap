// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer renders footerLink styles added 1`] = `
.emotion-16 {
  padding: 5rem 0 1rem;
  text-align: center;
  width: 100%;
}

.emotion-12 {
  list-style: none;
  margin: 0 0 1rem;
  padding: 0;
}

.emotion-2 {
  display: inline-block;
  padding: 0 1rem;
}

.emotion-0 {
  color: currentColor;
  -webkit-transition: color 400ms ease-in;
  transition: color 400ms ease-in;
}

<footer
  className="emotion-16 emotion-17"
>
  <nav
    className="emotion-14 emotion-15"
  >
    <ul
      className="emotion-12 emotion-13"
    >
      <li
        className="emotion-2 emotion-3"
      >
        <a
          className="emotion-0 emotion-1"
          href="/"
          onClick={[Function]}
          style={
            Object {
              "color": "#fff",
            }
          }
        >
          Home
        </a>
      </li>
      <li
        className="emotion-2 emotion-3"
      >
        <a
          className="emotion-0 emotion-1"
          href="/palette"
          onClick={[Function]}
          style={
            Object {
              "color": "#fff",
            }
          }
        >
          Palette
        </a>
      </li>
      <li
        className="emotion-2 emotion-3"
      >
        <a
          className="emotion-0 emotion-1"
          href="/about"
          onClick={[Function]}
          style={
            Object {
              "color": "#fff",
            }
          }
        >
          About
        </a>
      </li>
    </ul>
  </nav>
</footer>
`;

exports[`Footer renders without crashing 1`] = `
.emotion-16 {
  padding: 5rem 0 1rem;
  text-align: center;
  width: 100%;
}

.emotion-12 {
  list-style: none;
  margin: 0 0 1rem;
  padding: 0;
}

.emotion-2 {
  display: inline-block;
  padding: 0 1rem;
}

.emotion-0 {
  color: currentColor;
  -webkit-transition: color 400ms ease-in;
  transition: color 400ms ease-in;
}

<footer
  className="emotion-16 emotion-17"
>
  <nav
    className="emotion-14 emotion-15"
  >
    <ul
      className="emotion-12 emotion-13"
    >
      <li
        className="emotion-2 emotion-3"
      >
        <a
          className="emotion-0 emotion-1"
          href="/"
          onClick={[Function]}
        >
          Home
        </a>
      </li>
      <li
        className="emotion-2 emotion-3"
      >
        <a
          className="emotion-0 emotion-1"
          href="/palette"
          onClick={[Function]}
        >
          Palette
        </a>
      </li>
      <li
        className="emotion-2 emotion-3"
      >
        <a
          className="emotion-0 emotion-1"
          href="/about"
          onClick={[Function]}
        >
          About
        </a>
      </li>
    </ul>
  </nav>
</footer>
`;
